lib/modules/showImages.js

```diff
@@ -472,11 +472,15 @@ addModule('showImages', (module, moduleID) => {
 	}
 
 	function updateRevealedImages(type) {
-		for (const image of imageList) {
-			if ($(image).hasClass(type) || image.imageLink.expandOnViewAll) {
-				revealImage(image, findImageFilter(image.imageLink));
-			}
+		let images = imageList.filter(v => $(v).hasClass(type) || v.imageLink.expandOnViewAll);
+
+		if (RESUtils.pageType() === 'linklist') {
+			images = images.map(v => [$(v).closest('.thing').data('rank'), v])
```

Comment:
`Rank` only exists in `linklist`-pages, so an alternative could be to save the element number in the DOM when calling `checkElementForImage` and use that as basis for the sort.

Alternatively, change the way `imageList` is created and filled.